aws/table_aws_application_signals_service_level_objective.go

```diff
@@ -0,0 +1,174 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/service/applicationsignals"
+	cloudwatchlogsv1 "github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/applicationsignals/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+func tableAwsApplicationSignalsServiceLevelObjective(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_application_signals_service_level_objective",
+		Description: "AWS Application Signals Service Level Objective",
+		Get: &plugin.GetConfig{
+			Hydrate:    getApplicationSignalsServiceLevelObjective,
+			Tags:       map[string]string{"service": "application_signals", "action": "GetApplicationSignalsServiceLevelObjective"},
+			KeyColumns: plugin.AllColumns([]string{"arn", "name"}),
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listApplicationSignalsServiceLevelObjectives,
+			Tags:    map[string]string{"service": "application_signals", "action": "ListApplicationSignalsServiceLevelObjectives"},
+		},
+		HydrateConfig: []plugin.HydrateConfig{
+			{
+				Func: getApplicationSignalsServiceLevelObjective,
+				Tags: map[string]string{"service": "application_signals", "action": "GetApplicationSignalsServiceLevelObjective"},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(cloudwatchlogsv1.EndpointsID),
```

Comment:
Query, how do I identify regions supported by _this_ application_signals service

Reply:
Hello @johnlayton, 

Thank you for your contribution! 🎉 

In this plugin, we are currently using AWS Go SDK v1, specifically version `v1.51.19`. Unfortunately, in this version, the AWS **Application Signals** service is not yet available. After reviewing the AWS Go SDK v1 repository, I found that support for this service was added in version [v1.53.20](https://github.com/aws/aws-sdk-go/commit/b58ff6c4c01fd8d3ccea3958b2c50dc305b2fd22).

To use the `EndpointID` of the Application Signals service, we would need to update the AWS Go SDK v1 to `v1.53.20`. This would enable us to leverage the [EndpointID reference](https://github.com/aws/aws-sdk-go/blob/v1.55.3/service/applicationsignals/service.go#L34).

Alternatively, if we prefer not to update the AWS SDK v1, we could potentially use a hardcoded value (`application-signals`) to retrieve the supported regions for the Application Signals service. However, I’m not entirely certain if using a hardcoded value would fully resolve the issue in listing supported regions for the service.

Please let us know what works best for you. 

Thanks again!

Reply:
Great feedback.  Just to confirm (I am new to Go) the `go.mod` and `go.sum` changes need to be reverted except for adding `github.com/aws/aws-sdk-go-v2/service/applicationsignals v1.3.1`?

Reply:
Hello @johnlayton,

I noticed that the current package version we are using in the AWS plugin ([v1.27.0](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2@v1.27.0)) already supports the [applicationsignals](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/applicationsignals) service. Therefore, there’s no need to update AWS SDK V2 to support this table.

However, for the service endpoint reference, we may need to update AWS SDK V1 to version `v1.53.20`.

To resolve this, you can follow the steps outlined below:
1. Revert the [AWS SDK V2 to v1.27.0](https://github.com/turbot/steampipe-plugin-aws/pull/2291/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R9) by running the command: `go get github.com/aws/aws-sdk-go-v2@v1.27.0`.
2. Update AWS SDK V1 to `v1.53.20` to obtain the endpoint ID reference by running the command: `go get github.com/aws/aws-sdk-go@v1.53.20`.
3. Replace the hardcoded endpoint ID value [here](https://github.com/turbot/steampipe-plugin-aws/pull/2291/files#diff-292eaee8a49e4d7fa2ead26672c1eabb20f4d34d7b522619f9c80b7c2cf82120R35).
4. Make any necessary adjustments to the `aws_application_signals_service_level_objective` table, depending on the API behavior, if required.

Please let me know if you need any further clarification. 

Thanks!

Reply:
Hi.  I updated the dependencies as suggested.  I still see changes to `configsources` and `endpoints`, and the linting fails.  Is this expected or am I still missing something?

Reply:
Hello @johnlayton, thank you so much for your efforts and contribution! I’ve merged the PR into a branch.

The lint failure was caused by the use of the deprecated function [EndpointResolverWithOptionsFunc](https://github.com/turbot/steampipe-plugin-aws/blob/84b9d848f8c5ca2f87021bc04405f794c7ea931c/aws/service.go#L1864) with AWS SDK V2 version 1.30.5.

I'll look into resolving it.